### DISCUSSION TITLE
Activate app window on keyboard shortcut

### DISCUSSION THREAD

**Discussion by DmitryBorodiy** [AvaloniaUI/Avalonia#15369] [Q&A] [answered]
Hello, community!
Help me create class with Linux/Windows keyboard shortcut handling to activate window. For example my app is running and I known how to use topmost property, but I need to activate my app with keyboard shortcut and make my window Topmost = true.

**Comment by maxkatz6** [comment]
You can't handle global keyboard events, if your app isn't focused. If that's what you want.
You can call native APIs though https://github.com/AvaloniaUI/Avalonia/discussions/8823 (likely, to be available on Windows only, and some old X11 distros)

**Reply by DmitryBorodiy** [reply]
I've solved this problem with developing a background service on Windows that listens for events from the keyboard. I don't know yet how to implement this on Linux.
I need the app to appear on top of all windows when I press the keyboard shortcut.

**Reply by thevortexcloud** [reply, marked as answer]
You can take a look at Sharphook.

https://github.com/TolikPylypchuk/SharpHook

I can't say I have ever tried it. As I mentioned before you can poll the raw keyboard data, but you likely wont have permission to do that. Otherwise you can use x11 APIs.But the more modern way which is also not guaranteed to work since its up to desktops to implement is via the keyboard shortcut portals. Which seems to be implemented in KDE (not sure about other desktops).

**Comment by thevortexcloud** [comment]
Depending on your use case, it's also possible to just parse the keyboard data directly on Linux, which will work on X11 and Wayland without issue. It's just a file after all. But you need root/permission to read the file. It also contains binary data so you will need to convert it into something useful yourself.
